### PR TITLE
delay the inevitable a little longer and reenable python 2.7 support

### PR DESCRIPTION
**Pull request by pkienzle**

### CODE REVIEW

**Comment by andyfaff** [comment]
May I enquire why Python2.7 compatibility is required again?  Wouldn't it be better to do that in a dedicated branch, rather than implementing these changes?

**Comment by pkienzle** [comment]
We still need py27 because jenkins is build 4.x using py27.  This small patch is quicker than updating the build system.  So hold off on py3 only syntax until 4.3 is released or abandoned.

File: sasmodels/generate.py [hunk 1]
```diff
@@ -902,6 +899,10 @@ def _add_source(source, code, path, lineno=1):
     source.append('#line %d "%s"' % (lineno, path))
     source.append(code)
 
+def read_text(f):
```
Comment (krzywon) [inline sasmodels/generate.py:902]: Alternate solution that allows py3 to continue using pathlib:
```suggestion
# CRUFT: py2.7 support
try:
    from pathlib.Path import read_text
    from pathlib import Path
except ImportError:
    class Path:
        def __init__(self, f):
            self.f = f
        def read_text(self):
            with open(self.f) as fid:
                return fid.read()
```

File: setup.py [hunk 2]
```diff
@@ -33,14 +31,17 @@ def find_version(package):
 
 install_requires = ['numpy', 'scipy']
 
+with open('README.rst') as fid:
```
Comment (krzywon) [inline setup.py:34]: My other suggestion could work here too.

File: sasmodels/generate.py (hunk 1 above)
Comment (pkienzle) [reply sasmodels/generate.py:902]: Path is a class not a module, so you can't import from it.  You could maybe monkeypatch the backported pathlib to include read_text but probably not worthwhile.

**Comment by butlerpd** [comment]
Looks like minimal changes and seem reasonable to me.  Jeff tested in both 2.7 and 3.6 so will merge this

**Pull request merged**